### DISCUSSION TITLE
Can I force reencode h264 videos?  Acceptable video codecs settings

### DISCUSSION THREAD

**Discussion by SystemZ** [immich-app/immich#10049] [Q&A] [answered]
Hi, I got some h264 encoded 4K videos with high bitrate from my camera.
I would want to apply re-encoding even if it's already h264 for smooth playback on mobile connection.
I'm ok with more space taken this way.

I'm using Immich v1.105.1 and I tried to disable all acceptable video codecs but after saving settings it's enabled back.
I can enable or disable for example VP9 or AV1 but is seems that h264 is always selected by force.
Is this by design, some bug or I set something wrong?

<img width="314" alt="image" src="https://github.com/immich-app/immich/assets/1836094/01cc2643-6da5-4fb3-b060-4af7e2748592">

**Comment by mertalev** [comment, marked as answer]
The target video codec is always added as an accepted codec. It sounds like a different transcode policy might be what you're looking for. The bitrate-based policy will transcode anything above the max bitrate you set.

**Comment by SystemZ** [comment]
Thanks, it was it! 😎
I managed to set it to re-encode every video to 720p h264, even already h264.

VIDEO CODEC - h264
ACCEPTED VIDEO CODECS - select only H.264
TARGET RESOLUTION - 720p
TRANSCODE POLICY - All videos

**Reply by WhiteWoodenWall** [reply]
I was also having issues with my 4k h265 videos taken on a SONY a6700, where the videos would show as corrupted in both Firefox and the Android app. Setting the Transcode Policy to "Videos higher than target resolution or not in an accepted format" fixed it for me too!
